cmd/neofs-adm/internal/modules/morph/subnet.go

```diff
@@ -910,3 +802,405 @@ func init() {
+	}
+
+	cosigners, err := notaryCosigners(c, notaryHash, nnsCs, key, hash.Hash160(multisigScript))
+	if err != nil {
+		return fmt.Errorf("cosigners collecting: %w", err)
+	}
+
+	params, err := invocationParams(args...)
+	if err != nil {
+		return fmt.Errorf("invocation parameters prepararion: %w", err)
+	}
+
+	// make test invocation of the method
+	test, err := c.InvokeFunction(subnetHash, method, params, cosigners)
+	if err != nil {
+		return fmt.Errorf("test invocation: %w", err)
+	}
+
+	err = checkInvocationResults(test)
+	if err != nil {
+		return err
+	}
+
+	multisigAccount := &wallet.Account{
+		Contract: &wallet.Contract{
+			Script: multisigScript,
+		},
+	}
+
+	bc, err := c.GetBlockCount()
+	if err != nil {
+		return fmt.Errorf("blockchain height: %w", err)
+	}
+
+	signersNumber := uint8(len(alphabet)*2/3 + 1 + 1) // alphabet multisig + key signature
+
+	mainTx := &transaction.Transaction{
+		Nonce:           rand.Uint32(),
+		SystemFee:       test.GasConsumed,
+		ValidUntilBlock: bc + 100,
+		Script:          test.Script,
+		Attributes: []transaction.Attribute{
+			{
+				Type:  transaction.NotaryAssistedT,
+				Value: &transaction.NotaryAssisted{NKeys: signersNumber},
+			},
+		},
+		Signers: cosigners,
+	}
+
+	notaryFee, err := c.CalculateNotaryFee(signersNumber)
+	if err != nil {
+		return err
+	}
+
+	acc := wallet.NewAccountFromPrivateKey(&key)
+	aa := notaryAccounts(multisigAccount, acc)
+
+	err = c.AddNetworkFee(mainTx, notaryFee, aa...)
+	if err != nil {
+		return fmt.Errorf("notary network fee adding: %w", err)
+	}
+
+	mainTx.Scripts = notaryWitnesses(c, multisigAccount, acc, mainTx)
+
+	_, err = c.SignAndPushP2PNotaryRequest(mainTx,
+		[]byte{byte(opcode.RET)},
+		-1,
+		0,
+		40,
+		acc)
+	if err != nil {
+		return fmt.Errorf("sending notary request: %w", err)
+	}
+
+	return nil
+}
+
+func notaryCosigners(c Client, notaryHash util.Uint160, nnsCs *state.Contract,
+	key keys.PrivateKey, alphabetAccount util.Uint160) ([]transaction.Signer, error) {
+	proxyHash, err := nnsResolveHash(c, nnsCs.Hash, proxyContract+".neofs")
+	if err != nil {
+		return nil, fmt.Errorf("proxy hash resolving: %w", err)
+	}
+
+	return []transaction.Signer{
+		{
+			Account: proxyHash,
+			Scopes:  transaction.None,
+		},
+		{
+			Account: alphabetAccount,
+			Scopes:  transaction.Global,
+		},
+		{
+			Account: hash.Hash160(key.PublicKey().GetVerificationScript()),
+			Scopes:  transaction.Global,
+		},
+		{
+			Account: notaryHash,
+			Scopes:  transaction.None,
+		},
+	}, nil
+}
+
+func notaryAccounts(alphabet, acc *wallet.Account) []*wallet.Account {
+	aa := make([]*wallet.Account, 0, 4)
+
+	// proxy
+	aa = append(aa, &wallet.Account{
+		Contract: &wallet.Contract{
+			Deployed: true,
+		},
+	})
+
+	// alphabet
+	aa = append(aa, alphabet)
+
+	// caller's account
+	aa = append(aa, acc)
+
+	// last one is a placeholder for notary contract account
+	aa = append(aa, &wallet.Account{
+		Contract: &wallet.Contract{},
+	})
+
+	return aa
```

Comment:
Why not make it a `return []*wallet.Account{...}`, would be more readable IMO.

Reply:
agree, the only one reason i did so: comments readability

changed